pkg/network/cache/types.go

```diff
@@ -35,6 +35,7 @@ type DHCPConfig struct {
 	Mtu                 uint16
 	IPAMDisabled        bool
 	Gateway             net.IP
+	Subdomain           string
```

Comment:
Since the subdomain is part of the vmi spec, there is no need to store it in the cache in phase1 and read the value from the cache in phase2. Phase2 can read the subdomain directly from the spec.